src/hooks/useCreateGroupChannel.ts

```diff
@@ -49,8 +49,12 @@ export function useCreateGroupChannel(
           message: firstMessage,
         });
       }
-      // let's wait for a second to make sure channel is created & ready
-      await delay(1000);
+      let count = 0;
+      while (groupChannel.lastMessage == null && count < 30) {
+        await delay(100);
+        count += 1;
+      }
```

Comment:
nit;

```
async function waitForLastMessage(groupChannel, maxRetries = 30, retryInterval = 100) {
  let count = 0;
  while (groupChannel.lastMessage == null && count < maxRetries) {
    await delay(retryInterval);
    count++;
  }
  await delay(500);
}
```
How about having this logic separately and call it in here? 
```
// usage 
await waitForLastMessage(groupChannel);
```

Reply:
@AhyoungRyu Your suggestion is also good. Could you update it?

Reply:
Yeah sure will do.